File: problemSet/25A-IQ_Test.py
"""
Codeforces
25 A. IQ Test

@author yamaton
@date 2015-08-04
"""

# The description "evenness" is just terrible!
def solve(xs):
    odds = [i for i, x in enumerate(xs, 1) if x % 2 == 1]
    evens = [i for i, x in enumerate(xs, 1) if x % 2 == 0]
    if len(odds) == 1:
        return odds[0]
    elif len(evens) == 1:
        return evens[0]
    else:
        raise TypeError('wrong input!')


def main():
    n = int(input())
    xs = [int(i) for i in input().strip().split()]
    assert n == len(xs)
    result = solve(xs)
    print(result)


if __name__ == '__main__':
    main()

File: problemSet/25A_IQ_test.cpp
/*
 * Codeforces
 *
 * Problem 25 A. IQ test
 *
 * author: yamaton
 * date: 2015-11-26
 * 
 */
#include <algorithm>
#include <iostream>
#include <vector>
#include <iterator>

using namespace std;

bool even(int n) {
    return n % 2 == 0;
}


int solve2(vector<int> xs, int n) {
    auto even_count = count_if(xs.begin(), xs.end(), even);
    auto res = (even_count == 1) ? 
                find_if(xs.begin(), xs.end(), even) - xs.begin()
                : find_if_not(xs.begin(), xs.end(), even) - xs.begin();
    return res + 1;
}


int solve(vector<int> xs, int n) {
    int evens = 0;
    int pos_even, pos_odd;
    for(int i = 0; i < n; ++i) {
        if(xs[i] % 2 == 0) {
            evens++, pos_even = i + 1;
        } else {
            pos_odd = i + 1;
        }
    }
    return (evens == 1) ? pos_even : pos_odd;
}


int main() {
    int n;
    cin >> n;

    vector<int> xs(n);
    for(auto& x: xs) 
        cin >> x;

    cout << solve2(xs, n) << endl;
    return 0;
}